[MIG] account_cutoff_base: Migration to 16.0

dzungtran89 opened the pull request:
Needs
- https://github.com/OCA/account-closing/pull/212: for `pre-commit` fixes

Note:

- I removed the `groups` from the `company_id` field is to avoid violating the check by the native, see [here](https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/models/ir_ui_view.py#L3014)

Commit:
Initial check-in of my cut-off modules

Commit:
Coding style enhancements (osv.Model -> orm.Model, from . import xxx)

Commit:
Add _prepare functions to allow to inherit before create
Use assert
Remove .keys()
No space before colon
PEP8/Flake8 : getting closer to compliancy

Commit:
Forgot to move some code to the new _prepare function.

Commit:
Remove the unnecessary "and True or False" after 'xxx' not in context.get('type', '-')

Commit:
Reduce flake8 warnings (cut lines and comments, space after coma inside domains)
On account.account, type must be <> 'view' and <> 'closed'

Commit:
My christmas present to the PEP8 fundamentalists : there is not even a single warning left !
Add translation template files.

gurneyalex committed:
all modules unported to odoo 8.0

sbidoul committed:
[MIG] move account_cutoff_base and account_cutoff_prepaid out of unported

sbidoul committed:
[MIG] account_cutoff: set installable=True

Commit:
[MIG] account_cutoff_prepaid 8.0

sbidoul committed:
[IMP] account_cutoff_prepaid: add hooks

The signature of existing methods is preserved.

Commit:
[ADD][account_cutoff_base] Add multi-company record rule

yvaucher committed:
Add missing default oca icons

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

alexis-via committed:
Port account_cutoff_base and account_cutoff_prepaid to Odoo v9

Extract a new module account_invoice_start_end_dates from account_cutoff_prepaid

Commit:
Add some hooks that are in 8.0

pedrobaeza committed:
[MIG] Rename manifest files

alexis-via committed:
Take into account the remarks of Pedro:

Use triple double quotes for docstring
Replace <openerp> by <odoo> in XML
Remove <data> tags in XML
Remove POT files

Commit:
Migration of account_cutoff modules to 10.0

Commit:
[IMP] Add some tests to improve the code coverage

Commit:
[FIX] Bad attributes on m2m fields.

Commit:
[MIG] account_cutoff_base: Migration to 11.0

oca-travis committed:
[UPD] Update account_cutoff_base.pot

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (73 of 73 strings)

Translation: account-closing-11.0/account-closing-11.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-11-0/account-closing-11-0-account_cutoff_base/de/

Commit:
[MIG] account_cutoff_base: Migration to 12.0

Commit:
[12.0-mig-account] fixing review comments

oca-travis committed:
[UPD] Update account_cutoff_base.pot

sbidoul committed:
[FIX] remove titles from readme fragments

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-closing-12.0/account-closing-12.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-12-0/account-closing-12-0-account_cutoff_base/

OCA-git-bot committed:
[UPD] README.rst

Commit:
moved cutoff menu in submenu menu_finance_entries

OCA-git-bot committed:
account_cutoff_base 12.0.1.0.1

jimhoefnagels committed:
[IMP] account_cutoff_base: black, isort

jimhoefnagels committed:
[MIG] account_cutoff_base: Migration to 13.0

OCA-git-bot committed:
[UPD] README.rst

alexis-via committed:
Up-port improvements from v12

Fix bug in v13 (assert converted to raise without inverting the condition !)
Use button to switch to forecast mode.
Improve v13 migration: update record rules, defaulf method for company_id, company_id in domain

alexis-via committed:
Black and other pre-commit stuff

alexis-via committed:
FIX visibility of forecast button

Default value for cutoff date is end date of previous fiscal year

alexis-via committed:
Remove unused variable

OCA-git-bot committed:
[UPD] README.rst

alexis-via committed:
Add option move_partner

code is not required on account.analytic.account, so we can't use it in
list view

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_cutoff_base.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-closing-13.0/account-closing-13.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-13-0/account-closing-13-0-account_cutoff_base/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 68.3% (71 of 104 strings)

Translation: account-closing-13.0/account-closing-13.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-13-0/account-closing-13-0-account_cutoff_base/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (104 of 104 strings)

Translation: account-closing-13.0/account-closing-13.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-13-0/account-closing-13-0-account_cutoff_base/es_AR/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-closing-13.0/account-closing-13.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-13-0/account-closing-13-0-account_cutoff_base/

teewux committed:
[FIX] account_cutoff_base: res company

company.py should be loaded first due to used field
default_cutoff_move_partner on account_cutoff.py

alexis-via committed:
[MIG] account_cutoff_base to v14

Now include account_cutoff_accrual_base, as suggested in issue #133

oca-travis committed:
[UPD] Update account_cutoff_base.pot

OCA-git-bot committed:
[UPD] README.rst

rousseldenis committed:
[14.0][FIX] account_cutoff_base: Move domains where they are used

Avoid ValidationError on res.company views update as company_id field
does not exist there

oca-travis committed:
[UPD] Update account_cutoff_base.pot

OCA-git-bot committed:
account_cutoff_base 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/

alexis-via committed:
account_cutoff_base: widget="text" on name field (same as on account.move)

Add optional="show/hide" on some fields of cutoff lines

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 45.5% (61 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

OCA-git-bot committed:
account_cutoff_base 14.0.1.2.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 51.4% (69 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 76.8% (103 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 87.3% (117 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (134 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (134 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (134 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

AntonioMariaVigliotti committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (134 of 134 strings)

Translation: account-closing-14.0/account-closing-14.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-14-0/account-closing-14-0-account_cutoff_base/it/

yankinmax committed:
[IMP] : black, isort, prettier

yankinmax committed:
[IMP] account_cutoff_base: pylint

yankinmax committed:
[MIG] account_cutoff_base: Migration to 15.0

alexis-via committed:
account_cutoff_base: improve tree view of account.cutoff

Commit:
[UPD] Update account_cutoff_base.pot

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (134 of 134 strings)

Translation: account-closing-15.0/account-closing-15.0-account_cutoff_base
Translate-URL: https://translation.odoo-community.org/projects/account-closing-15-0/account-closing-15-0-account_cutoff_base/es_AR/

dzungtran89 committed:
[IMP] account_cutoff_base: black, isort, prettier

dzungtran89 committed:
[MIG] account_cutoff_base: Migration to 16.0

ivantodorovich commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -0,0 +1,79 @@
+    company_currency_id = fields.Many2one(
+        related="parent_id.company_currency_id",
+        string="Company Currency",
+        readonly=True,
```
readonly is redundant here, in case you want to remove it

ivantodorovich commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -0,0 +1,79 @@
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", readonly=True)
+    quantity = fields.Float(digits="Product Unit of Measure", readonly=True)
+    price_unit = fields.Float(
```
This could be a Monetary field

alexis-via replied:
No. price_unit are not monetary fields for decimal precision reasons, cf https://github.com/odoo/odoo/blob/16.0/addons/sale/models/sale_order_line.py#L139
A price_unit can have a decimal precision higher than the currency precision.

ivantodorovich commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -0,0 +1,79 @@
+        help="Price per unit (discount included) without taxes in the default "
+        "unit of measure of the product in the currency of the 'Currency' field.",
+    )
+    price_origin = fields.Char(readonly=True)
```
I fail to understand the purpose of this field, and why it is a `Char` but it's name suggests a price (thus it should be `Float` or `Monetary`)

Is this field even used? If so, what for? Should this be commented in the `help` attribute?

alexis-via replied:
It's a small note field that indicate where the price has been taken from. So it contains an invoice number, a PO number or an SO number.

ivantodorovich commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -0,0 +1,79 @@
+    price_origin = fields.Char(readonly=True)
+    origin_move_line_id = fields.Many2one(
+        "account.move.line", string="Origin Journal Item", readonly=True
+    )  # Old name: move_line_id
```
Maybe remove the `# Old name:` comments.
Seem like they're outdated by now

ivantodorovich commented on account_cutoff_base/models/account_cutoff_line.py:
```diff
@@ -0,0 +1,79 @@
+    origin_move_date = fields.Date(
+        related="origin_move_line_id.move_id.date", string="Origin Journal Entry Date"
+    )  # old name: move_date
+    account_id = fields.Many2one(
```
IMO, this model should have `_check_company_auto = True`, and a check_company on this `account_id` field.
(Maybe also other fields, if relevant)

ivantodorovich commented on account_cutoff_base/models/account_cutoff_tax_line.py:
```diff
@@ -0,0 +1,51 @@
+        ondelete="cascade",
+        required=True,
+    )
+    tax_id = fields.Many2one("account.tax", string="Tax", required=True)
```
`check_company` maybe?

ivantodorovich reviewed (approved):
Non-blocking comments

dzungtran89 commented:
@ivantodorovich , thank you for you comments, the PR has been updated

About the field `price_origin`, I'm not sure about its purpose tbh, it was initially added in the [commit](https://github.com/OCA/account-closing/pull/213/commits/a3fc49ed252697dc89bbf2cedc21535abaff89c5#diff-a30853050e1420bf74baaaba68936d3e32217851cc7aa701d50053ce3af9ee58R431)

alexis-via commented:
I'm preparing a PR on this PR, to take into account my comments and make a few improvements

alexis-via committed:
account_cutoff_base: forward-port source_move_state from v14

Revert price_unit to fields.Float() and price_origin to fields.Char()
Forward port post_cutoff_move on res.company from v14
Add ACLs for account readonly group
Remove duplicate-xml-fields, cf https://github.com/OCA/maintainer-tools/issues/558
Add name_get() on account.cutoff
Rename field move_label to move_ref
Update code to be able to add additional states (e.g. 'forecast' state introduced account_cutoff_start_end_dates)
Cleanup and improve views

alexis-via commented:
Here it is: https://github.com/dzungtran89/account-closing/pull/1

dzungtran89 committed:
Merge pull request #1 from akretion/16.0-account_cutoff_base-fix-improve

[16.0] account_cutoff_base: improve your OCA PR

alexis-via commented:
/ocabot merge nobump

simahawk commented:
/ocabot migration account_cutoff_base

simahawk commented:
@dzungtran89 next time, please, remove merge commits if you integrate other PRs. Thanks.

dzungtran89 commented:
hi @simahawk, well noted with thanks
